src/site/content/en/fast/lazy-loading-images/index.md

```diff
@@ -0,0 +1,302 @@
+most compatible across browsers, modern browsers offer a more performant and
+efficient way to do the work of checking element visibility via [the
+intersection observer API](/web/updates/2016/04/intersectionobserver).
+
+{% Aside %}
+  Intersection observer is not supported in all browsers.
+  If compatibility across browsers is crucial,
+  be sure to read [the next section](#using_event_handlers_the_most_compatible_way),
+  which shows you how to lazy-load images using less performant (but more compatible!) scroll and resize event handlers.
+{% endAside %}
+
+Intersection observer is easier to use and read than code relying on various
+event handlers, because you only need to register an observer to watch
+elements rather than writing tedious element visibility detection code. All
+that's left to do is to decide what to do when an element is visible.
+Let's assume this basic markup pattern for your lazily loaded `<img>` elements:
+
+```html
+<img class="lazy" src="placeholder-image.jpg" data-src="image-to-lazy-load-1x.jpg" data-srcset="image-to-lazy-load-2x.jpg 2x, image-to-lazy-load-1x.jpg 1x" alt="I'm an image!">
+```
+
+There are three relevant pieces of this markup that you should focus on:
+
+1. The `class` attribute, which is what you'll select the element with in
+JavaScript.
+2. The `src` attribute, which references a placeholder image that will appear when
+the page first loads.
+3. The `data-src` and `data-srcset` attributes, which are placeholder attributes
+containing the URL for the image you'll load once the element is in the viewport.
+
+Now let's see how to use intersection observer in JavaScript to lazy-load
+images using this markup pattern:
+
+```javascript
+document.addEventListener("DOMContentLoaded", function() {
+  var lazyImages = [].slice.call(document.querySelectorAll("img.lazy"));
+
+  if ("IntersectionObserver" in window) {
+    let lazyImageObserver = new IntersectionObserver(function(entries, observer) {
+      entries.forEach(function(entry) {
+        if (entry.isIntersecting) {
+          let lazyImage = entry.target;
+          lazyImage.src = lazyImage.dataset.src;
+          lazyImage.srcset = lazyImage.dataset.srcset;
+          lazyImage.classList.remove("lazy");
+          lazyImageObserver.unobserve(lazyImage);
+        }
+      });
+    });
+
+    lazyImages.forEach(function(lazyImage) {
+      lazyImageObserver.observe(lazyImage);
+    });
+  } else {
+    // Possibly fall back to a more compatible method here
+  }
+});
+```
+
+On the document's `DOMContentLoaded` event, this script queries the DOM for all
+`<img>` elements with a class of `lazy`. If intersection observer is available,
+create a new observer that runs a callback when `img.lazy` elements enter the
+viewport.
+
+<div class="glitch-embed-wrap" style="height: 420px; width: 100%;">
+  <iframe
+    src="https://glitch.com/embed/#!/embed/lazy-intersection-observer?path=index.html&previewSize=100"
```

Comment:
I think there's an option (something you append to the Glitch URL) to choose whether the preview shows the code or the app. It's not a big deal, but I think in this case it's better to show the code by default because the app is blank initially. (And it seems like lazy-loading is triggered only when you scroll inside the Glitch iframe?)

Reply:
I think setting `previewSize` to 0 will only show the code.